cpp/benchmarks/reduction/scan_structs.cpp

```diff
@@ -45,16 +47,31 @@ static void nvbench_structs_scan(nvbench::state& state)
   auto [null_mask, null_count] = create_random_null_mask(size, null_probability);
   auto const input             = cudf::make_structs_column(
     size, std::move(data_table->release()), null_count, std::move(null_mask));
+  auto input_view = input->view();
 
   auto const agg         = cudf::make_min_aggregation<cudf::scan_aggregation>();
   auto const null_policy = static_cast<cudf::null_policy>(state.get_int64("null_policy"));
   auto const stream      = cudf::get_default_stream();
 
   state.set_cuda_stream(nvbench::make_cuda_stream_view(stream.value()));
-  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
-    auto const result = cudf::detail::scan_inclusive(
-      *input, *agg, null_policy, stream, rmm::mr::get_current_device_resource());
-  });
+  int64_t result_size = 0;
+  state.exec(nvbench::exec_tag::sync | nvbench::exec_tag::timer,
+             [&](nvbench::launch& launch, auto& timer) {
+               timer.start();
+               auto const result = cudf::detail::scan_inclusive(
+                 input_view, *agg, null_policy, stream, rmm::mr::get_current_device_resource());
+               timer.stop();
+
+               // Estimating the result size will launch a kernel. Do not include it in measuring
+               // time.
+               result_size += estimate_size(result->view());
+             });
```

Comment:
This looks strange to me. I'm not sure we need to accumulate the result sizes since they should always be the same.
Maybe something like this (not sure if this compiles):
```suggestion
  std::unique_ptr<cudf::column> result = nullptr;
  state.exec(nvbench::exec_tag::sync | nvbench::exec_tag::timer,
             [&](nvbench::launch& launch, auto& timer) {
               result = cudf::detail::scan_inclusive(
                 input_view, *agg, null_policy, stream, rmm::mr::get_current_device_resource());
             });
  auto const result_size = estimate_size(result->view());
```

Reply:
Oh.. I did not realize that the function given to `state.exec()` is executed multiple times. I guess I should have expected that though as benchmark tools are supposed to do that. Yes, you are right. The `result_size` should not be accumulated. Now the result shows the valid throughputs as below. Thanks for catching this!

```
|        T        | null_probability | data_size | Samples |  CPU Time  | Noise  |  GPU Time  | Noise |  Elem/s  | GlobalMem BW | BWUtil |
|-----------------|------------------|-----------|---------|------------|--------|------------|-------|----------|--------------|--------|
|             I32 |                0 |     10000 |  17968x |  31.892 us | 14.76% |  27.839 us | 2.73% |  87.074M | 707.473 MB/s |  0.09% |
|             I32 |              0.1 |     10000 |  17520x |  32.590 us | 14.46% |  28.554 us | 3.25% |  84.893M | 689.752 MB/s |  0.09% |
|             I32 |              0.5 |     10000 |  17616x |  32.508 us | 14.79% |  28.386 us | 3.05% |  85.395M | 693.836 MB/s |  0.09% |
...
```